File: javascript/matrices/1-SumOfDiagonalValues.js
// Write a program that fills a 3x3 matrix with values entered by the user
// and displays the sum of the main diagonal values.

const prompt = require('prompt-sync')({ sigint: true });

const matrix = [];
let sum = 0;

// Prompt a user for matrix values
for (let i = 0; i < 3; i++) {
  matrix[i] = [];
  for (let j = 0; j < 3; j++) {
    matrix[i][j] = parseInt(prompt(`Enter a value for matrix[${i}][${j}]: `));
    if (i === j) {
      sum += matrix[i][j]; // Add values on the main diagonal
    }
  }
}

// display matrix and sum of main diagonal
console.log('Matrix:', matrix);
console.log('The sum of main diagonal values:', sum);

File: python/matrices/1-SumOfDiagonalValues.py
# Write a program that fills a 3x3 matrix with values entered by the user
# and displays a sum of the main diagonal values.

# Initialize a 3x3 matrix
matrix = [[0, 0, 0],
          [0, 0, 0],
          [0, 0, 0]]

# Prompt the user to enter the values for the matrix
print("Enter values for the 3x3 matrix:")

for i in range(3):
  for j in range(3):
    value = int(input("{}, {} ".format(i, j)))
    matrix[i][j] = value

# Calculate the sum of the main diagonal values
sum_diagonal = 0

for i in range(3):
  sum_diagonal += matrix[i][i]

# Display the matrix
print("Matrix:")
for row in matrix:
  print(row)

# Display the sum of the main diagonal values
print("The sum of main diagonal values:", sum_diagonal)